lib/ansible/modules/windows/win_snmp.ps1

```diff
@@ -0,0 +1,116 @@
+#!powershell
+
+# Copyright: (c) 2017, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.Legacy
+
+$params      = Parse-Args -arguments $args -supports_check_mode $true;
+$check_mode  = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+$managers    = Get-AnsibleParam -obj $params -name "permitted_managers"  -type "list" -default @()
```

Comment:
Having this the default means that someone can only specify communities with `action: set` and this will wipe out the managers, and vice versa. Wouldn't it be better to have the default as `$null` and get people to explicitly state `managers: []` or `community_strings: []` if they want to set it to empty. We can always add an example to show how this can be done. If we do the above, the module should then fail if either one is not set. If we don't change this behaviour then it should be stated in the docs.